### PR TITLE
Remove ref from the text_filter

### PR DESCRIPTION
**Pull request by igor-dv**
## What I did
I've removed `ref` from the text_filter (which was added by me earlier)

### CODE REVIEW

File: lib/ui/src/modules/ui/components/left_panel/text_filter.js [hunk 1]
```diff
@@ -48,33 +50,28 @@ export default class TextFilter extends React.Component {
     super(props);
 
     this.state = {
-      query: '',
+      query: props.text || defaultTextValue,
```
Comment (Hypnosphi) [inline lib/ui/src/modules/ui/components/left_panel/text_filter.js]: it's already a defaultProp

File: lib/ui/src/modules/ui/components/left_panel/text_filter.js [hunk 2]
```diff
@@ -48,33 +50,28 @@ export default class TextFilter extends React.Component {
...
 
     this.onChange = this.onChange.bind(this);
     this.fireOnClear = this.fireOnClear.bind(this);
-    this.setInputRef = this.setInputRef.bind(this);
     this.changeFilter = debounce(this.changeFilter, debounceFilterChangeTimeout);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.text !== this.state.query) {
+      this.setState({ query: nextProps.text || defaultTextValue });
```
Comment (Hypnosphi) [inline lib/ui/src/modules/ui/components/left_panel/text_filter.js]: same here

**Review by Hypnosphi — CHANGES_REQUESTED** [review]

File: lib/ui/src/modules/ui/components/left_panel/text_filter.js (hunk 1 above)
Comment (igor-dv) [inline lib/ui/src/modules/ui/components/left_panel/text_filter.js]: Yeah, but I've got an error when `null` was passed..

Comment (Hypnosphi) [inline lib/ui/src/modules/ui/components/left_panel/text_filter.js]: Let's do that fallback just once then — directly in the input's value prop

**Review by Hypnosphi — APPROVED** [review]

**Pull request merged**